builtin/providers/aws/resource_aws_instance.go

```diff
@@ -986,7 +997,15 @@ func buildAwsInstanceOpts(
 		}
 	}
 
-	if hasSubnet && associatePublicIPAddress {
+	if hasNetworkInterface {
+		ni := &ec2.InstanceNetworkInterfaceSpecification{
+			NetworkInterfaceId: aws.String(networkInterfaceID),
+			DeviceIndex:        aws.Int64(int64(0)),
+			SubnetId:           aws.String(subnetID),
```

Comment:
This will need some more work I just wanted to get it out there before
going on travel.
Subnet_id cannot be specified with network interface.
I'll have to look into the others. My guess is that they should be be
specified since they'll be defined on the ENI.
On Wed, Dec 9, 2015 at 11:12 AM Clint <notifications@github.com> wrote:

> In builtin/providers/aws/resource_aws_instance.go
> <https://github.com/hashicorp/terraform/pull/4231#discussion_r47120749>:
>
> > @@ -986,7 +997,15 @@ func buildAwsInstanceOpts(
> >  		}
> >  	}
> >
> > -	if hasSubnet && associatePublicIPAddress {
> > +	if hasNetworkInterface {
> > +		ni := &ec2.InstanceNetworkInterfaceSpecification{
> > +			NetworkInterfaceId: aws.String(networkInterfaceID),
> > +			DeviceIndex:        aws.Int64(int64(0)),
> > +			SubnetId:           aws.String(subnetID),
>
> We're also don't yet consider private_ip or vpc_security_group_ids, are
> those coming, or do they not apply here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hashicorp/terraform/pull/4231/files#r47120749>.
>